### PR TITLE
Star Wars Characters project

### PR DESCRIPTION
**Pull request by sheyna**
# Week 1 HW Submission

Please fill out the information below in order to complete your assignment. Feel free to update this comment later if necessary.

* Comfort rating on this assignment (1-5): 3
I had already figured out route in last week's assignment, but I had problems with getting the tests to take the data from the api in one case. And for some reason most of my PropTypes didn't seem to take.

* Completion rating on this assignment (complete/incomplete): Look at my PropTypes and PersonCard.test.js, those would be the problem areas.
 
* Github handles of code I've reviewed (if any):

### CODE REVIEW

File: star-wars-films/src/PersonCard/PersonCard.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+describe('PersonCard component', () => {
+  it('should render', () => {
+    const mockData = {
+      "name": "Luke Skywalker",
+      "url": "https://swapi.co/api/people/1/"
+    };
+    const component = ReactTestRenderer.create(
+      <MemoryRouter>
+        <div>
+          <PersonCard people={mockData} />;
```
Comment (sheyna) [inline star-wars-films/src/PersonCard/PersonCard.test.js]: I tried multiple times to pass in data from an actual api call here. It never worked. In the end I mocked the data. I'd be curious to see how to do this with an actual api call.

File: star-wars-films/src/PersonDetail/PersonDetail.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  getFilmList(films) {
+    const filmNames = ["A New Hope", "The Empire Strikes Back", "Return of the Jedi", "The Phantom Menace", "Attack of the Clones", "Revenge of the Sith", "The Force Awakens", "The Last Jedi"];
+    let finalFilmList = '';
+    for (let i = 0; i < films.length; i++){
+      const url = films[i];
+      let filmNum = url.split('/').reverse()[1];
+      filmNum = filmNum - 1;
+      finalFilmList += filmNames[filmNum];
+      if (i < (films.length - 1)) {
+        finalFilmList += ", ";
```
Comment (sheyna) [inline star-wars-films/src/PersonDetail/PersonDetail.js]: I had to create a list this more sloppy way as I couldn't get map or forEach to work, see comment in this file.

File: star-wars-films/src/PersonDetail/PersonDetail.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  // // Can't do below as return statement in getFilmList
+  // // tells me that finalFilmList.map is not a function;
+  // // tried finalFilmList.forEach and got the same error.
+  //
+  // {finalFilmList.map((finalFilmList, idx) => {
+  //    return <span>{finalFilmList}, </span>
+  // })}
+
+  getHomePlanet(planet) {
+    const planetNames = ["Tatooine","Alderaan","Yavin IV","Hoth","Dagobah","Bespin","Endor","Naboo","Coruscant", "Kamino", "Geonosis", "Utapau", "Mustafar", "Kashyyyk", "Polis Massa", "Mygeeto", "Felucia", "Cato Neimoidia", "Saleucami", "Stewjon", "Eriadu", "Corellia", "Rodia", "Nal Hutta", "Dantooine", "Bestine IV", "Ord Mantell", "unknown", "Trandosha", "Socorro", "Mon Cala", "Chandrila", "Sullust", "Toydaria", "Malastare", "Dathomir", "Ryloth", "Aleen Minor", "Vulpter", "Troiken", "Tund", "Haruun Kal", "Cerea", "Glee Anselm", "Iridonia", "Tholoth", "Iktotch", "Quermia", "Dorin", "Champala", "Mirial", "Serenno", "Concord Dawn","Zolan","Ojom","Skako","Muunilinst","Shili","Kalee","Umbara","Jakku"]
```
Comment (sheyna) [inline star-wars-films/src/PersonDetail/PersonDetail.js]: Like with the film data above. I kind of faked this without doing a second API call. I'd be curious how to do second call inside this file.

File: star-wars-films/src/PersonDetail/PersonDetail.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            <div className="styleized-font">{name}</div>
+          </div>
+        </section>
+      </div>
+    )
+  }
+}
+
+// these PropTypes do not appear to work:
+
```
Comment (sheyna) [inline star-wars-films/src/PersonDetail/PersonDetail.js]: I could never gets these PropTypes to work. I could change them to any data type and I'd never get an error message.

File: star-wars-films/src/PersonCard/PersonCard.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        <div className="container">
+          <p className="styleized-font"><span>{people.name}</span></p>
+          <h1>{people.name}</h1>
+        </div>
+        </Link>
+      </div>
+    )
+  }
+}
+
```
Comment (sheyna) [inline star-wars-films/src/PersonCard/PersonCard.js]: I could never gets these PropTypes to work. I could change them to any data type and I'd never get an error message.

File: star-wars-films/src/Home/Home.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        </section>
+        <div className="pagination">
+          {this.state.page > 1 ? this.getPrevButton() : null}
+          {this.state.page < 9 ? this.getNextButton() : null}
+        </div>
+      </div>
+    )
+  }
+}
+
```
Comment (sheyna) [inline star-wars-films/src/Home/Home.js]: This is the only place I was able to get my PropTypes to work. If I changed "number" to any other data type, as expected, I'd get an error.

**Review by sheyna — COMMENTED** [review]
My own comments about my code

File: star-wars-films/src/Home/Home.js (hunk 6 above)
Comment (irenab1) [reply star-wars-films/src/Home/Home.js]: page is the only thing is this.props.   The rest is state, and I don't think you can check it with propTypes

File: star-wars-films/src/PersonCard/PersonCard.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+// these PropTypes do not appear to work:
+
+PersonCard.propTypes = {
+  people: PropTypes.object.isRequired,
+  people: PropTypes.shape({
+    url: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired
+  })
```
Comment (irenab1) [inline star-wars-films/src/PersonCard/PersonCard.js]: I think you'd remove line 35 and add
}).isRequired on line 39

File: star-wars-films/src/PersonDetail/PersonDetail.js (hunk 2 above)
Comment (irenab1) [reply star-wars-films/src/PersonDetail/PersonDetail.js]: In Home.js for state i would've added films: [].   In Home.js componentDidMount add fetch(https://swapi.co/api/films/) and put it into state films[].  In PersonDetail.js you have person.films array of urls.   You can loop thru that array and search each element in state films[] by url

File: star-wars-films/src/Home/Home.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    );
+  }
+
+  render() {
+    const people = this.state.people;
+    return (
+      <div className="home">
+        <section className="page-content">
+          {people.map((people, idx) => {
+            return <PersonCard key={idx} people={people} />;
```
Comment (irenab1) [inline star-wars-films/src/Home/Home.js]: confusing naming.  Some of "people" should be renamed "person"

File: star-wars-films/src/PersonCard/PersonCard.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import './PersonCard.css';
+import {
+  Link
+} from "react-router-dom";
+import PropTypes from 'prop-types';
+
+class PersonCard extends Component {
+
+  render() {
+    const people = this.props.people;
```
Comment (irenab1) [inline star-wars-films/src/PersonCard/PersonCard.js]: s.b. person, not people

File: star-wars-films/src/PersonCard/PersonCard.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        </Link>
+      </div>
+    )
+  }
+}
+
+// these PropTypes do not appear to work:
+
+PersonCard.propTypes = {
+  people: PropTypes.object.isRequired,
```
Comment (irenab1) [inline star-wars-films/src/PersonCard/PersonCard.js]: I'd remove line 35

File: star-wars-films/src/PersonDetail/PersonDetail.js (hunk 3 above)
Comment (irenab1) [reply star-wars-films/src/PersonDetail/PersonDetail.js]: I'd add planets[] to state in home.js and get that list in home.js componentDidMount.

File: star-wars-films/src/PersonDetail/PersonDetail.js (hunk 4 above)
Comment (irenab1) [reply star-wars-films/src/PersonDetail/PersonDetail.js]: Isn't person in state rather than in this.props ?

**Review by irenab1 — APPROVED** [review]

**Pull request closed without merging**